src/app/components/FilterPopup/FilterPopup.jsx

```diff
@@ -92,12 +100,26 @@ class FilterPopup extends React.Component {
   }
 
   onFilterClick(filterId, filter) {
-    const selectedFilters = this.state.selectedFilters;
+    const {
+      provisionalSelectedFilters,
+      selectedFilters,
+    } = this.state;
 
+    const clickedFilters = {
+      materialType: _union(provisionalSelectedFilters.materialType, selectedFilters.materialType),
+      language: _union(provisionalSelectedFilters.language, selectedFilters.language),
+      dateAfter: selectedFilters.dateAfter,
+      dateBefore: selectedFilters.dateBefore,
+    };
     if (filter.selected) {
-      selectedFilters[filterId].push(filter);
+      clickedFilters[filterId].push(filter);
       trackDiscovery('Filters - Select', `${filterId} - ${filter.label}`);
     } else {
+      clickedFilters[filterId] =
+        _reject(
+          clickedFilters[filterId],
+          selectedFilter => selectedFilter.value === filter.value,
+        );
       selectedFilters[filterId] =
```

Comment:
If a filter is already selected, we need to remove it from the state. But the same filter is also tracked provisionally so it needs to be removed from both arrays.